diff -Naur dmai_2_20_00_15_ori/packages/ti/sdo/dmai/Makefile dmai_2_20_00_15_new/packages/ti/sdo/dmai/Makefile
--- dmai_2_20_00_15_ori/packages/ti/sdo/dmai/Makefile	2011-11-20 13:29:23.660224911 +0530
+++ dmai_2_20_00_15_new/packages/ti/sdo/dmai/Makefile	2011-11-20 13:35:26.408224419 +0530
@@ -110,7 +110,7 @@
 GCC_CPP_FLAGS	+= $(CPP_FLAGS) -I$(LINUXKERNEL_INSTALL_DIR)/include -I$(LINUXKERNEL_INSTALL_DIR)/arch/arm/include -I$(LINUXKERNEL_INSTALL_DIR)/arch/arm/$(PLATFORM_ARCH)/include $(HAVE_AUTOCONFIG) -I$(CMEM_INSTALL_DIR)/packages -I$(LINK_INSTALL_DIR) -DDmai_BuildOs_linux -Dxdc_target_types__=gnu/targets/std.h -I$(LINUXLIBS_INSTALL_DIR)/include -Dxdc_target_name__=arm/GCArmv5T
 C64P_CPP_FLAGS	+= $(CPP_FLAGS) -I$(BIOS_INSTALL_DIR)/packages/ti/bios/include -I$(BIOS_INSTALL_DIR)/packages -DDmai_BuildOs_bios -Dxdc_target__='<ti/targets/std.h>' -I$(CODEGEN_INSTALL_DIR)/include -mv6400+ -pdse225 -eo$(PLATFORM_DEVICE).o$(EXT)
 
-GCC_C_FLAGS	+= $(C_FLAGS) -Wall -Werror
+GCC_C_FLAGS	+= $(C_FLAGS) -Wall  
 C64P_C_FLAGS	+= $(C_FLAGS)
 
 GNU_AR_FLAGS	+= $(AR_FLAGS)
